extension/headerssetterextension/extension.go

```diff
@@ -48,7 +50,22 @@ func newHeadersSetterExtension(cfg *Config) (auth.Client, error) {
 				Key: *header.FromContext,
 			}
 		}
-		headers = append(headers, Header{key: *header.Key, source: s})
+
+		var a action.Action
+		switch header.Action {
+		case INSERT:
+			a = action.Insert{Key: *header.Key}
+		case UPSERT:
+			a = action.Upsert{Key: *header.Key}
+		case UPDATE:
+			a = action.Update{Key: *header.Key}
+		case DELETE:
+			a = action.Delete{Key: *header.Key}
+		default:
+			a = action.Upsert{Key: *header.Key}
+			logger.Warn("Please set the header action explicitly")
```

Comment:
I would have added the default action directly to the `createDefaultConfig` and let the remaining of the code require upsert. It should be easier to change in the future.

Also, if we are planning on requiring the action to be explicitly set in a future version, we should already let users know right now. Like: `The action was not provided, using 'upsert'. In future versions, we'll require this to be explicitly set`

Reply:
> I would have added the default action directly to the createDefaultConfig and let the remaining of the code require upsert. It should be easier to change in the future.

The config is a list of `HeaderConfig`, so I am not sure whether we can set the default with `createDefaultConfig` then. We can set the default action in `Validate` but there is no access to the logger and validate mutates the config which is probably ok, but I preferred to put it in the `newHeadersSetterExtension` factory method.